File: Answer/2126.摧毁小行星.cs
/*
 * @lc app=leetcode.cn id=2126 lang=csharp
 *
 * [2126] 摧毁小行星
 */

// @lc code=start
using System;

public class Solution
{
    public bool AsteroidsDestroyed(int mass, int[] asteroids)
    {
        Array.Sort(asteroids);
        long currentMass = mass;
        foreach (int asteroid in asteroids)
        {
            if (currentMass < asteroid)
            {
                return false;
            }
            currentMass += asteroid;
        }
        return true;
    }
}
// @lc code=end

File: Answer/2126.摧毁小行星.py
#
# @lc app=leetcode.cn id=2126 lang=python3
#
# [2126] 摧毁小行星
#

# @lc code=start
from typing import List


class Solution:
    def asteroidsDestroyed(self, mass: int, asteroids: List[int]) -> bool:
        asteroids.sort()
        current_mass = mass
        for asteroid in asteroids:
            if current_mass < asteroid:
                return False
            current_mass += asteroid
        return True


# @lc code=end
